gradoop-temporal/src/main/java/org/gradoop/temporal/model/api/TemporalGraphOperators.java

```diff
@@ -44,14 +44,29 @@
   //----------------------------------------------------------------------------
 
   /**
-   * Extracts a snapshot of this temporal graph using a given temporal predicate.
-   * This will calculate the subgraph induced by the predicate.
+   * Extracts a snapshot of this temporal graph using a given temporal predicate. The predicate is applied
+   * on the valid time dimension by default. To use the transaction time dimension, use
+   * {@link TemporalGraphOperators#snapshot(TemporalPredicate, TimeDimension)} instead.
+   * This operator will calculate the subgraph induced by the predicate.
    *
-   * @param predicate the temporal predicate to apply
+   * @param predicate the temporal predicate to apply on the valid times
    * @return the snapshot as a temporal graph
    */
   default TemporalGraph snapshot(TemporalPredicate predicate) {
-    return callForGraph(new Snapshot(Objects.requireNonNull(predicate)));
+    return snapshot(predicate, TimeDimension.VALID_TIME);
+  }
+
+  /**
+   * Extracts a snapshot of this temporal graph using a given temporal predicate. The predicate is applied
+   * on the given time dimension.
+   * This operator will calculate the subgraph induced by the predicate.
+   *
+   * @param predicate the temporal predicate to apply
+   * @param dimension the dimension that is used
+   * @return the snapshot as a temporal graph
+   */
+  default TemporalGraph snapshot(TemporalPredicate predicate, TimeDimension dimension) {
+    return callForGraph(new Snapshot(Objects.requireNonNull(predicate), Objects.requireNonNull(dimension)));
```

Comment:
I think the `null` check should be part of the `Snapshot` constructor, so I think you can remove this here